pallets/simple-nft/src/tests.rs

```diff
@@ -320,51 +344,77 @@ fn it_fails_for_destroying_multiple_tokens_as_other_signer() {
-        // asset old token hasn't changed
+        // assert old token hasn't changed
         assert_eq!(token_2, SimpleNFTModule::tokens_by_id(2));
     });
 }
 
 #[test]
 fn it_fails_for_destroying_single_burnt_token() {
     new_test_ext().execute_with(|| {
-        SimpleNFTModule::run_process(Origin::signed(1), Vec::new(), vec![(1, 42)]).unwrap();
+        let metadata0 = BTreeMap::from_iter(vec![(0, H256::zero())]);
+        let metadata1 = BTreeMap::from_iter(vec![(0, H256::zero())]);
+        SimpleNFTModule::run_process(Origin::signed(1), Vec::new(), vec![(1, metadata0.clone())]).unwrap();
         SimpleNFTModule::run_process(Origin::signed(1), vec![1], Vec::new()).unwrap();
         // get old token
         let token = SimpleNFTModule::tokens_by_id(1);
         // Try to destroy token as incorrect user
         assert_err!(
-            SimpleNFTModule::run_process(Origin::signed(1), vec![1], vec![(1, 43)]),
+            SimpleNFTModule::run_process(Origin::signed(1), vec![1], vec![(1, metadata1.clone())]),
             Error::<Test>::AlreadyBurnt
         );
         // assert no more tokens were created
         assert_eq!(SimpleNFTModule::last_token(), 1);
-        // asset old token hasn't changed
+        // assert old token hasn't changed
         assert_eq!(token, SimpleNFTModule::tokens_by_id(1));
     });
 }
 
 #[test]
 fn it_fails_for_destroying_multiple_tokens_with_burnt_token() {
     new_test_ext().execute_with(|| {
-        SimpleNFTModule::run_process(Origin::signed(1), Vec::new(), vec![(1, 42), (1, 43)]).unwrap();
+        let metadata0 = BTreeMap::from_iter(vec![(0, H256::zero())]);
+        let metadata1 = BTreeMap::from_iter(vec![(0, H256::zero())]);
+        let metadata2 = BTreeMap::from_iter(vec![(0, H256::zero())]);
+        SimpleNFTModule::run_process(Origin::signed(1), Vec::new(), vec![(1, metadata0.clone()), (1, metadata1.clone())]).unwrap();
         SimpleNFTModule::run_process(Origin::signed(1), vec![1], Vec::new()).unwrap();
         // get old token
         let token_1 = SimpleNFTModule::tokens_by_id(1);
         // get old token
         let token_2 = SimpleNFTModule::tokens_by_id(2);
         // Try to destroy token as incorrect user
         assert_err!(
-            SimpleNFTModule::run_process(Origin::signed(1), vec![1, 2], vec![(1, 44)]),
+            SimpleNFTModule::run_process(Origin::signed(1), vec![1, 2], vec![(1, metadata2.clone())]),
             Error::<Test>::AlreadyBurnt
         );
         // assert no more tokens were created
         assert_eq!(SimpleNFTModule::last_token(), 2);
-        // asset old token hasn't changed
+        // assert old token hasn't changed
         assert_eq!(token_1, SimpleNFTModule::tokens_by_id(1));
-        // asset old token hasn't changed
+        // assert old token hasn't changed
         assert_eq!(token_2, SimpleNFTModule::tokens_by_id(2));
     });
 }
+
+
+#[test]
+fn it_fails_for_creating_single_token_with_too_many_metadata_items() {
```

Comment:
Need another test covering multiple metadata values on a token and perhaps one where multiple outputs have different numbers of valid metadata items

Reply:
Added a multiple metadata values on a single token test and I think I've added the second one?

Reply:
https://github.com/digicatapult/vitalam-node/pull/8/commits/305ff62fd14cc0ac3e0a7e6cf5a5be35589dd5a9